cpp/src/jni/parquet/parquet_reader.h

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef PARQUET_READER_H
+#define PARQUET_READER_H
+
+#include <arrow/record_batch.h>
+#include <parquet/arrow/reader.h>
+#include <parquet/file_reader.h>
+#include <parquet/properties.h>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+#include "jni/parquet/file_connector.h"
+#include "jni/parquet/hdfs_connector.h"
+
+namespace jni {
+namespace parquet {
+
+class ParquetReader {
+ public:
+  explicit ParquetReader(std::string path);
+  ::arrow::Status initialize(std::vector<int>& column_indices,
```

Comment:
these should either be const references or passed via pointer if a mutation is expected (and moved to the end of the function call).

Reply:
@emkornfield , fixed, thanks